In [1]:
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
df= pd.read_csv('https://raw.githubusercontent.com/MinCiencia/Datos-COVID19/master/output/producto81/vacunacion_comuna_edad_1eraDosis.csv')
df.head()

Region  Codigo region                          Comuna  \
0  Arica y Parinacota             15                           Arica   
1  Arica y Parinacota             15                       Camarones   
2  Arica y Parinacota             15                   General Lagos   
3  Arica y Parinacota             15                           Putre   
4  Arica y Parinacota             15  Desconocido Arica y Parinacota   

   Codigo comuna  Poblacion   15   16   17     18     19  ...      71      72  \
0        15101.0   247552.0  0.0  0.0  0.0  499.0  699.0  ...  1375.0  1328.0   
1        15102.0     1233.0  0.0  0.0  0.0    3.0    2.0  ...     3.0    22.0   
2        15202.0      810.0  0.0  0.0  0.0    1.0    0.0  ...     2.0     7.0   
3        15201.0     2515.0  0.0  0.0  0.0    0.0    2.0  ...     9.0    11.0   
4            NaN        NaN  NaN  NaN  NaN    NaN    NaN  ...     NaN     NaN   

       73      74      75     76      77     78     79      80  
0  1304.0  1218.0  1156.0  995.0  1058.0  901.0  747.0  5085.0  
1    11.0    17.0     6.0    5.0     6.0    5.0    6.0    27.0  
2     2.0     4.0     4.0    2.0     3.0    4.0    2.0    10.0  
3     2.0    12.0    10.0    4.0     9.0    9.0    8.0    51.0  
4     NaN     NaN     NaN    NaN     NaN    NaN    NaN     NaN  

[5 rows x 71 columns]

a continuacion lo que hace loc es dejarme solo las lineas que contengan metropolitana en su columna de region

In [3]:
df=df.loc[df['Region']=='Metropolitana']
df.head(55)

Region  Codigo region                     Comuna  Codigo comuna  \
88   Metropolitana             13                      Alhue        13502.0   
89   Metropolitana             13                       Buin        13402.0   
90   Metropolitana             13            Calera de Tango        13403.0   
91   Metropolitana             13                  Cerrillos        13102.0   
92   Metropolitana             13                Cerro Navia        13103.0   
93   Metropolitana             13                     Colina        13301.0   
94   Metropolitana             13                   Conchali        13104.0   
95   Metropolitana             13                   Curacavi        13503.0   
96   Metropolitana             13                  El Bosque        13105.0   
97   Metropolitana             13                   El Monte        13602.0   
98   Metropolitana             13           Estacion Central        13106.0   
99   Metropolitana             13                 Huechuraba        13107.0   
100  Metropolitana             13              Independencia        13108.0   
101  Metropolitana             13              Isla de Maipo        13603.0   
102  Metropolitana             13                La Cisterna        13109.0   
103  Metropolitana             13                 La Florida        13110.0   
104  Metropolitana             13                  La Granja        13111.0   
105  Metropolitana             13                 La Pintana        13112.0   
106  Metropolitana             13                   La Reina        13113.0   
107  Metropolitana             13                      Lampa        13302.0   
108  Metropolitana             13                 Las Condes        13114.0   
109  Metropolitana             13               Lo Barnechea        13115.0   
110  Metropolitana             13                  Lo Espejo        13116.0   
111  Metropolitana             13                   Lo Prado        13117.0   
112  Metropolitana             13                      Macul        13118.0   
113  Metropolitana             13                      Maipu        13119.0   
114  Metropolitana             13                Maria Pinto        13504.0   
115  Metropolitana             13                  Melipilla        13501.0   
116  Metropolitana             13                      Nunoa        13120.0   
117  Metropolitana             13              Padre Hurtado        13604.0   
118  Metropolitana             13                      Paine        13404.0   
119  Metropolitana             13        Pedro Aguirre Cerda        13121.0   
120  Metropolitana             13                   Penaflor        13605.0   
121  Metropolitana             13                  Penalolen        13122.0   
122  Metropolitana             13                     Pirque        13202.0   
123  Metropolitana             13                Providencia        13123.0   
124  Metropolitana             13                   Pudahuel        13124.0   
125  Metropolitana             13                Puente Alto        13201.0   
126  Metropolitana             13                  Quilicura        13125.0   
127  Metropolitana             13              Quinta Normal        13126.0   
128  Metropolitana             13                   Recoleta        13127.0   
129  Metropolitana             13                      Renca        13128.0   
130  Metropolitana             13               San Bernardo        13401.0   
131  Metropolitana             13                San Joaquin        13129.0   
132  Metropolitana             13          San Jose de Maipo        13203.0   
133  Metropolitana             13                 San Miguel        13130.0   
134  Metropolitana             13                  San Pedro        13505.0   
135  Metropolitana             13                  San Ramon        13131.0   
136  Metropolitana             13                   Santiago        13101.0   
137  Metropolitana             13                  Talagante  

In [4]:
rq = pd.read_csv('https://raw.githubusercontent.com/leoferres/gitiele356b/main/data/hdi.csv')
rq

comuna       LEI        II        EI    idh       HDI
0    13101  0.791672  0.873137  0.924099  0.807  0.861222
1    13102  0.782817  0.800343  0.713499  0.743  0.764616
2    13103  0.752447  0.750382  0.657961  0.683  0.718874
3    13104  0.784872  0.760697  0.705426  0.707  0.749584
4    13105  0.777236  0.780019  0.673876  0.711  0.742015
5    13106  0.790810  0.784378  0.729145  0.735  0.767604
6    13107  0.726330  0.775631  0.641707  0.737  0.712375
7    13108  0.821551  0.751105  0.722493  0.709  0.763935
8    13109  0.856570  0.835740  0.860235  0.775  0.850780
9    13110  0.773094  0.824255  0.791155  0.804  0.795888
10   13111  0.763582  0.751721  0.668837  0.689  0.726793
11   13112  0.673794  0.758585  0.642574  0.679  0.689951
12   13113  0.890647  0.948908  0.968223  0.883  0.935337
13   13114  0.909774  0.971056  0.988014  0.933  0.955683
14   13115  0.831115  0.921621  0.813262  0.912  0.854046
15   13116  0.759407  0.784503  0.690043  0.657  0.743558
16   13117  0.787707  0.775519  0.697989  0.715  0.752665
17   13118  0.831334  0.842243  0.841346  0.806  0.838293
18   13119  0.768126  0.823406  0.808770  0.782  0.799755
19   13120  0.900363  0.947737  0.993361  0.860  0.946392
20   13121  0.815597  0.767565  0.733783  0.708  0.771589
21   13122  0.749867  0.790376  0.660819  0.743  0.731645
22   13123  0.917316  0.948546  1.044039  0.911  0.968496
23   13124  0.708344  0.764565  0.723853  0.735  0.731874
24   13125  0.667144  0.838734  0.763492  0.782  0.753152
25   13126  0.813982  0.794761  0.686444  0.723  0.762931
26   13127  0.775766  0.742308  0.672305  0.697  0.728831
27   13128  0.725183  0.782950  0.674829  0.709  0.726315
28   13129  0.814219  0.815018  0.747089  0.719  0.791456
29   13130  0.855578  0.851691  0.816075  0.765  0.840925
30   13131  0.767104  0.772013  0.605335  0.679  0.710381
31   13132  0.914831  0.996899  1.078779  0.949  0.994584
32   13201  0.725925  0.773923  0.721415  0.773  0.740045
33   13202  0.749975  0.847558  0.735982  0.807  0.776297
34   13203  0.789578  0.828198  0.654495  0.759  0.753607
35   13301  0.667378  0.787350  0.625280    NaN  0.690036
36   13302  0.683453  0.787269  0.688481    NaN  0.718193
37   13303  0.816336  0.775896  0.629003    NaN  0.735826
38   13401  0.743463  0.762822  0.676397  0.712  0.726599
39   13402  0.742570  0.822340  0.737301    NaN  0.766439
40   13403  0.801109  0.760228  0.601466    NaN  0.715509
41   13404  0.738969  0.776218  0.655095    NaN  0.721614
42   13501  0.787502  0.739201  0.642315    NaN  0.720423
43   13502  0.876923  0.756470  0.617345    NaN  0.742609
44   13503  0.811913  0.766193  0.647025    NaN  0.738340
45   13504  0.809846  0.756324  0.617808    NaN  0.723305
46   13505  0.819107  0.690242  0.556089    NaN  0.679979
47   13601  0.794968  0.802522  0.686845    NaN  0.759548
48   13602  0.740793  0.766459  0.679887    NaN  0.728127
49   13603  0.784525  0.808623  0.736952    NaN  0.776123
50   13604  0.764730  0.738680  0.659196  0.728  0.719437
51   13605  0.785732  0.801915  0.698606    NaN  0.760697

In [ ]:
pb = pd.read_csv('https://raw.githubusercontent.com/leoferres/gitiele356b/main/data/pobXedadCenso.csv')
pb.head()

In [ ]:
pb.dtypes

In [ ]:
drq =  rq[rq.duplicated()].shape
dpb= pb[pb.duplicated()].shape
ddf = df[df.duplicated()].shape
print(ddf)
print(dpb)
print(drq)

Arreglamos los asteriscos del dataset de poblacion(pb)
https://www.geeksforgeeks.org/replace-nan-values-with-zeros-in-pandas-dataframe/

In [ ]:
pb= pb.replace('*',0)
pb.head()

In [ ]:
pb['EDAD_0A5']=pb['EDAD_0A5'].astype(str).astype(int)
pb['EDAD_6A14']=pb['EDAD_6A14'].astype(str).astype(int)
pb['EDAD_15A64']=pb['EDAD_15A64'].astype(str).astype(int)
pb['EDAD_65YMAS']=pb['EDAD_65YMAS'].astype(str).astype(int)
pb.dtypes

pb2 = pb[(pb['COMUNA']== 13502) | (pb['COMUNA']== 13402) | (pb['COMUNA']== 13403)] 
pb2

pb3= pb2.sum(axis=0)
pb3

In [ ]:
pb = pb.groupby('COMUNA').sum().reset_index()
pb = pb.rename(columns={'COMUNA':'Codigo_comuna'})
pb

https://www.geeksforgeeks.org/python-pandas-dataframe-groupby/

In [ ]:
df = df.dropna()
vac = df.loc[:,'Comuna':'Codigo comuna']
vac = vac.reset_index()
vac =vac.drop(['index'],axis=1)
vac

In [ ]:
vac.dtypes

In [ ]:
vac=vac.rename(columns={'Comuna':'Nombre_comuna','Codigo comuna':'Codigo_comuna'})

In [ ]:
merged = pd.merge(vac, pb, on='Codigo_comuna')
merged

In [ ]:
rq=rq.rename(columns={'comuna':'Codigo_comuna'})


In [ ]:
merged2 = pd.merge(merged,rq, on='Codigo_comuna')
merged2

In [ ]:
rq

In [ ]:
vac['%VAC_EDAD_0A5 '] = vac.iloc[:, 5:56].sum(axis=1)
vac